[sps30] remove delay

mrtoy-me opened the pull request:
# What does this implement/fix?

remove delay and replace with set_time. Just a slight change in logic so there is a short 20ms delay regardless of the state of "result" 

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Code quality improvements to existing code or addition of tests
- [ ] Other

**Related issue or feature (if applicable):**
One component in list for issue https://github.com/esphome/backlog/issues/38 "Minimize delay calls as much as possible"

- fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):**

- esphome/esphome-docs#<esphome-docs PR number goes here>

## Test Environment

- [ ] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040
- [ ] BK72xx
- [ ] RTL87xx
- [ ] nRF52840

## Example entry for `config.yaml`:

```yaml
# Example config.yaml

```

## Checklist:
  - [ ] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [ ] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

mrtoy-me committed:
remove delay

pre-commit-ci-lite[bot] committed:
[pre-commit.ci lite] apply automatic fixes

mrtoy-me committed:
fix context

pre-commit-ci-lite[bot] committed:
[pre-commit.ci lite] apply automatic fixes

mrtoy-me committed:
fix

mrtoy-me committed:
Merge branch 'sps30_remove_delay' of https://github.com/mrtoy-me/esphome into sps30_remove_delay

swoboda1337 commented on esphome/components/sps30/sps30.cpp:
```diff
@@ -45,24 +45,25 @@ void SPS30Component::setup() {
+      this->skipped_data_read_cycles_ = 0;
+      this->start_continuous_measurement_();
+    });
   });
```
May also want to set a startup complete flag here too?

mrtoy-me committed:
add setup complete flag

used to stop contents of update running before setup is complete also will prevent update running if component is mark failed by setup
